gluonnlp/model/char_encoder.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Convolutional character encoder."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+__all__ = ['CharacterEncoder']
+
+from mxnet import gluon, nd
+from mxnet.gluon import nn
+
+from .highway import Highway
+
+
+class CharacterEncoder(gluon.Block):
+    r"""Convolutional character encoder.
+
+    We implement the convolutional character encoder proposed in the following work::
+
+        @inproceedings{kim2016character,
+         title={Character-Aware Neural Language Models.},
+         author={Kim, Yoon and Jernite, Yacine and Sontag, David and Rush, Alexander M},
+         booktitle={AAAI},
+         pages={2741--2749},
+         year={2016}
+        }
+
+    Parameters
+    ----------
+    embed_size : int
+        The input dimension to the encoder.
+    num_filters: int
```

Comment:
It seems the original implementation sets `num_filters` as a function of `ngram_filter_sizes`. They use `min(200, 50*filter_size)`. Would it make sense to support setting different `num_filters`for different filter sizes?

Reply:
Now we allow users to specify the number of filters they want to apply according to different filter sizes. Thanks